android/src/ch/opengis/qfield/QFieldActivity.java

```diff
@@ -45,285 +43,305 @@
+                    if (!dir.isDirectory() && !dir.mkdirs())
+                        throw new FileNotFoundException("Failed to ensure directory: " + dir.getAbsolutePath());
+                    if (ze.isDirectory())
+                        continue;
+                    FileOutputStream fout = new FileOutputStream(file);
+                    try {
+                        while ((count = zis.read(buffer)) != -1)
+                            fout.write(buffer, 0, count);
+                    } finally {
+                        fout.close();
+                    }
+                    currentSize = is.available();
+
+                    // 108 instead of 100 beacuse the is.available() is not accurate on large files and with
+                    // 108 we have a 100% when the unzip is finished
+                    percent = 108 - currentSize / totalSize * 100;
+                    publishProgress((int) percent);
+
+                    Log.d(QtTAG, "File unzipped: " + file.getPath());
+                    Log.d(QtTAG, "Exists: " + file.exists());
+                    Log.d(QtTAG, "Percent: " + percent);
+                }
+            } finally {
+                zis.close();
+            }
+        }
+
+        protected void onPreExecute() {
+            unpackingDialog.show();
+
+            // create symlink
+            // alias paths for storage dir (/sdcard or similar)
+            String storagePathAlias = getFilesDir() + "/storage";
+            String storagePath = Environment.getExternalStorageDirectory()
+                .getAbsolutePath();
+
+            if (mExternalStorageAvailable) {
+                if (mExternalStorageWriteable) {
+                    String pathAlias;
+                    String externalFilesDir = getExternalFilesDir(null).getAbsolutePath();
+                    String filesDir = getFilesDir().getAbsolutePath();
+
+                    // put the share files to externalFilesDir
+                    pathAlias = filesDir + "/share";
+                    mShareDir = externalFilesDir + "/share";
+                    new File(mShareDir).mkdir();
+                    makeSymlink(mShareDir, pathAlias);
+
+                    // put .qgis to externalFilesDir
+                    pathAlias = filesDir + "/.qgis2";
+                    mDotQgis2Dir = externalFilesDir + "/.qgis2";
+                    new File(mDotQgis2Dir).mkdir();
+                    makeSymlink(mDotQgis2Dir, pathAlias);
+                } else {
+                    storagePathAlias = storagePathAlias + "ReadOnly";
+                }
+                makeSymlink(storagePath, storagePathAlias);
+            }
+        }
+
+        protected void onProgressUpdate(Integer... progress) {
+            super.onProgressUpdate(progress);
+            progressBar.setProgress(progress[0]);
+
+            if (progress[0] >= 60) {
+                unpackingMessage.setText(getString(R.string.unpacking_msg_following));
```

Comment:
the text will be updated each time, we don't care?

Reply:
I don't think it's an issue.